### PR TITLE
Replace deprecated functions

### PR DESCRIPTION
**Pull request by noxdafox**
Add support for Elixir >= 1.9.

Signed-off-by: Matteo Cafasso <noxdafox@gmail.com>

### CODE REVIEW

File: lib/mix/tasks/archive/build.all.ex [hunk 1]
```diff
@@ -61,7 +61,7 @@ defmodule Mix.Tasks.Archive.Build.All do
     destination = Mix.Archive.Build.Helpers.destination(opts)
     Mix.Tasks.Archive.Build.Deps.build_archives(opts)
 
-    archive_name = Mix.Local.name_for(:archive, Mix.Project.config)
+    archive_name = Mix.Local.name_for(:archives, Mix.Project.config)
```
Comment (noxdafox) [inline lib/mix/tasks/archive/build.all.ex:64]: This was changed in commit https://github.com/elixir-lang/elixir/commit/a94911881d4c0424fe3eaf154aa0d87cc5163b51. If we want to keep backwards compatibility, we need something a bit more elegant.

Comment (lukebakken) [reply lib/mix/tasks/archive/build.all.ex:64]: We don't build with anything earlier than Elixir 1.8.2, I believe.

Comment (lukebakken) [reply lib/mix/tasks/archive/build.all.ex:64]: Even with 1.8.2 we don't use `mix_task_archive_deps` anymore.

(cc @dumbbell @michaelklishin)

Comment (michaelklishin) [reply lib/mix/tasks/archive/build.all.ex:64]: Dropping Elixir 1.8 support is fine with me.

**Comment by lukebakken** [comment]
@noxdafox if you have a second to review commit 78b5fff378cb09d018a23f60a3082f3802ca9654 that would be great.

**Comment by noxdafox** [comment]
LGTM

**Pull request merged**